pkg/providers/instancetype/types.go

```diff
@@ -203,9 +203,10 @@ func memory(ctx context.Context, info *ec2.InstanceTypeInfo) *resource.Quantity
 }
 
 // Setting ephemeral-storage to be either the default value, what is defined in blockDeviceMappings, or the combined size of local store volumes.
+// nolint:gocyclo
 func ephemeralStorage(info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily, nodeClass *v1beta1.EC2NodeClass) *resource.Quantity {
 	// If local store disks have been configured for node ephemeral-storage, use the total size of the disks.
-	if lo.FromPtr(nodeClass.Spec.InstanceStorePolicy) == v1beta1.InstanceStorePolicyRAID0 {
+	if *nodeClass.Spec.AMIFamily == v1beta1.AMIFamilyAL2023 || lo.FromPtr(nodeClass.Spec.InstanceStorePolicy) == v1beta1.InstanceStorePolicyRAID0 {
```

Comment:
What was the output of the conversation with the AL2023 team? This means that we default in Karpenter to using RAID0 when it's not set. Are we planning to allow users to disable this in the future? If that's the case, do we know how we are going to do that?

Reply:
They are adding back the ability to specify, I'm working on making this change right now.

Reply:
https://github.com/awslabs/amazon-eks-ami/pull/1688